vispy/scene/widgets/grid.py

```diff
@@ -495,6 +495,11 @@ def _update_child_widget_dim(self):
             else:
                 y = np.sum(value_vectorized(self._height_grid[col][0:row]))
 
+            # Ensure we only update the widgets when a change occurs. Allow for
+            # floating point discrepencies in the comparisons.
+            if abs(widget.size[0] - width) < 1e-4 and abs(widget.size[1] - height) < 1e-4 and \
+                abs(widget.pos[0] - x) < 1e-4 and abs(widget.pos[1] - y) < 1e-4:
+                continue
```

Comment:
Couldn't (shouldn't?) this check be done inside the widget?

Reply:
Yes I think that would be better, it is more general then - I will make the change